java/ql/integration-tests/posix-only/kotlin/jvmoverloads_flow/test.kt

```diff
@@ -0,0 +1,78 @@
+fun getString() = "Hello world"
+
+fun source() = "tainted"
+
+fun sink(s: String) { }
+
+object Test {
+
+  @JvmOverloads @JvmStatic
+  fun taintSuppliedAsDefault(before: Int, s: String = source(), after: Int) { sink(s) }
+  
+  @JvmOverloads @JvmStatic
+  fun noTaintByDefault(before: Int, s: String = "no taint", after: Int, after2: Int = 1) { sink(s) }
+
+}
+
+public class Test2 {
+
+  companion object {
+
+    @JvmOverloads @JvmStatic
+    fun taintSuppliedAsDefaultStatic(before: Int, s: String = source(), after: Int) { sink(s) }
+
+    @JvmOverloads @JvmStatic
+    fun noTaintByDefaultStatic(before: Int, s: String = "no taint", after: Int, after2: Int = 1) { sink(s) }
+
+  }
+
+  @JvmOverloads
+  fun taintSuppliedAsDefault(before: Int, s: String = source(), after: Int) { sink(s) }
+
+  @JvmOverloads
+  fun noTaintByDefault(before: Int, s: String = "no taint", after: Int, after2: Int = 1) { sink(s) }
+
+}
+
+public class GenericTest<T> {
+
+  @JvmOverloads
+  fun taintSuppliedAsDefault(before: T, s: String = source(), after: T) { sink(s) }
+
+  @JvmOverloads
+  fun noTaintByDefault(before: T, s: String = "no taint", after: T, after2: Int = 1) { sink(s) }
+
+}
+
+public class ConstructorTaintsByDefault @JvmOverloads constructor(before: Int, s: String = source(), after: Int) {
+
+  init {
+    sink(s)
+  }
+
+}
+
+public class ConstructorDoesNotTaintByDefault @JvmOverloads constructor(before: Int, s: String = "no taint", after: Int, after2: Int = 1) {
+
+  init {
+    sink(s)
+  }
+   
+}
+
+public class GenericConstructorTaintsByDefault<T> @JvmOverloads constructor(before: T, s: String = source(), after: T) {  
+
+  init {
+    sink(s)
+  }
+   
+}
+ 
+public class GenericConstructorDoesNotTaintByDefault<T> @JvmOverloads constructor(before: T, s: String = "no taint", after: T, after2: T? = null) { 
+
+  init {
+    sink(s)
+  }
+
+}
+
```

Comment:
There's no test case added for any of the above methods being called from Kotlin. And as we discussed it on Slack, I think the calls are not working as expected.

Adding the following test case results in no new findings, but I think two new flows should be found.
```
class TestClass {
    companion object {
        @JvmOverloads @JvmStatic
        fun noTaintByDefault1(s1: String = "no taint", s2: String = "no taint") { sink(s2) }
    }

    fun noTaintByDefault2(s1: String = "no taint", s2: String = "no taint") { sink(s2) }

    fun fn() {
        noTaintByDefault1(s2 = source())
        noTaintByDefault2(s2 = source())
    }
}
```

Reply:
Indeed, I'm intentionally handling only the JvmOverloads / Java interop aspect here and will follow up to address calls from within other Kotlin code